crates/ide-db/src/search.rs

```diff
@@ -910,3 +919,10 @@ fn is_name_ref_in_import(name_ref: &ast::NameRef) -> bool {
         .and_then(|it| it.parent_path().top_path().syntax().parent())
         .map_or(false, |it| it.kind() == SyntaxKind::USE_TREE)
 }
+
+fn is_name_ref_in_test(sema: &Semantics<'_, RootDatabase>, name_ref: &ast::NameRef) -> bool {
+    name_ref.syntax().ancestors().any(|node| match ast::Fn::cast(node) {
+        Some(it) => sema.to_def(&it).map_or(false, |func| func.is_test(sema.db)),
+        None => false,
+    })
+}
```

Comment:
This won't work if the name ref is in a macro expansion as we only ascend the expanded tree, not the expansions.

The bigger issue though is that this will probably make searching quite a bit slower the more hits we have as we ascend the tree for every hit. (especially if we consider macro expansions as well). Unsure what we can do about that though, given we don't expand `#[test]` right now (but skip it) we can't really query the `HirFileId` for whether we are in a `test` expansion (which would then make this trivial). Expanding `#[test]` has its own downsides again though which is why we aren't doing that atm.

Reply:
I have tried to apply `Find All References` over `ide_assists::tests::check_assist`, which have 1684 refs when `excludeTests` set to `false`, and 113 refs when `excludeTests` set to `true`.  I feel there is no obvious difference in the response latency before and after the modification in this PR.

Reply:
Fair, I guess we do only run this on actual matches which shouldn't be too massive